internal/arc/arc_query_tx.go

```diff
@@ -0,0 +1,86 @@
+package arc
+
+import (
+	"context"
+	"fmt"
+	"io"
+	"strings"
+
+	"github.com/bitcoin-sv/go-broadcast-client/config"
+	"github.com/bitcoin-sv/go-broadcast-client/internal/decoder"
+	"github.com/bitcoin-sv/go-broadcast-client/internal/httpclient"
+	"github.com/bitcoin-sv/go-broadcast-client/models"
+	"github.com/bitcoin-sv/go-broadcast-client/shared"
+)
+
+var ErrMissingHash = fmt.Errorf("missing tx hash")
+
+func (a *ArcClient) QueryTransaction(ctx context.Context, txID string) (*models.QueryTxResponse, error) {
+
+	if a == nil {
+		return nil, shared.ErrClientUndefined
+	}
+
+	result, err := queryTransaction(ctx, a, txID)
+	if err != nil {
+		return nil, err
+	}
+
+	return result, nil
+}
+
+// queryTransaction will fire the HTTP request to retrieve the tx status and details
+func queryTransaction(ctx context.Context, arc *ArcClient, txHash string) (*models.QueryTxResponse, error) {
+	sb := strings.Builder{}
+	sb.WriteString(arc.apiURL + config.ArcQueryTxRoute + txHash)
```

Comment:
Why do you use builder here?

Reply:
Will be done in latest commit - resolved.
Moved to single variable.